nxc/protocols/smb/proto_args.py

```diff
@@ -2,6 +2,8 @@ def proto_args(parser, std_parser, module_parser):
         smb_parser = parser.add_parser("smb", help="own stuff using SMB", parents=[std_parser, module_parser])
         smb_parser.add_argument("-H", "--hash", metavar="HASH", dest="hash", nargs="+", default=[],
                                 help="NTLM hash(es) or file(s) containing NTLM hashes")
+        smb_parser.add_argument("--delegate", action="store", help="Impersonate user with S4U2Self + S4U2Proxy")
+        smb_parser.add_argument("--self", dest='no_s4u2proxy', action="store_true", help="Only do S4U2Self, no S4U2Proxy (use with delegate)")
```

Comment:
I think for using this, `--delegation` should be required. See ldap protoargs, "no_smb_arg"

Reply:
good idea